src/components/inputs/ColorSelect2/ColorSelect2.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useReducer, useRef, useEffect } from 'react';
+import { parseToHsl } from 'polished';
+
+import { Wrapper } from './ColorSelect2.style';
+import { Props } from './ColorSelect2.types';
+import { State, reducer } from './ColorSelect2.state';
+
+import { ColorInputs } from './Inputs';
+import { ColorSelectInput } from './ColorSelect2Input';
+import { ColorSelectPicker } from './ColorSelect2Picker';
+import { Presets, Props as PresetsProps } from './Presets';
+
+import { PopOver } from '../../PopOver/PopOver';
+import {
+  withIris,
+  MinorComponent,
+  useOutsideClick,
+} from '../../../utils';
+import { colorSpaces } from '../../../color';
+
+export const ColorSelect2 = withIris<
+  HTMLInputElement,
+  Props,
+  { Presets: MinorComponent<PresetsProps> }
+>(ColorSelectComponent);
+
+ColorSelect2.Presets = Presets;
+
```

Comment:
Nit: Let's just start putting descriptions of what these components are/how it's meant to be used and what the eventual intention of it is in a commented block.

Reply:
Added a comment in the ColorSelect2/Presets component

Reply:
OK sorry, got it :)
fixed